### ISSUE TITLE
Graphite checks are backwards

### ISSUE THREAD

**Issue by NickG123** [arachnys/cabot#255]
The documentation and help text for graphite checks state "A Graphite check will fail if any datapoint in the Metric would cause the expression [Metric] [Check type] [Value] to be true."  However, it seems that the reverse is true.  For example, the below check is currently failing for me:
![failingcheck](https://cloud.githubusercontent.com/assets/8438951/9585408/b7529e0a-4fe4-11e5-94f8-441c46015f2e.png)
The value is currently 92, while my check is "Less than or equal 90"
If the value drops below 90, the check begins to pass.
Either the documentation and help text needs to be updated to:  "A Graphite check will fail if any datapoint in the Metric would cause the expression  [Value] [Check type] [Metric] to be true." or the code needs to be changed to reflect the help text.  Personally I think it is more logical to change the code.

It's possible that this is the root cause of issues #234 and #129

**Comment by dbuxton** [comment]
I don't think that this holds in general - not sure what's up with your check here I must admit.

Can you put the JSON from Graphite into a gist (with anything sensitive redacted, obviously) and we can have a look?

**Comment by NickG123** [comment]
Took a brief look at the code, and it seems backwards to me:

                    if self.check_type == '<':
                        metric_failed = not last_value < float(self.value)
                    elif self.check_type == '<=':
                        metric_failed = not last_value <= float(self.value)
                    elif self.check_type == '>':
                        metric_failed = not last_value > float(self.value)
                    elif self.check_type == '>=':
                        metric_failed = not last_value >= float(self.value)
                    elif self.check_type == '==':
                        metric_failed = not last_value == float(self.value)

In this case self.check_type = '<=', last_value = 92, and self.value = 90 so we have:
metric_failed = not last_value <= float(self.value)
metric_failed = not 92 <= 90
metric_failed = not (false)
metric_failed = true

seems like a mix up of the less than / greater than cases, or an accidental use of not.

Just in case, here is the JSON from my metric:
https://gist.github.com/NickG123/57536100d39cafc47c8a
Essentially it is just 92 across the board.

**Comment by dbuxton** [comment]
Oops, that's pretty embarrassing. It looks like the version we have running in production doesn't have this, because it's using this code:

https://github.com/arachnys/cabot/blob/43ae71a986b44b42073a540727515df79a2f5158/cabot/cabotapp/models.py

The changes appear to have been introduced by this:

https://github.com/arachnys/cabot/commit/41f8f4e5a30151eb4be5ade72da3e591a8171d17#diff-65a212ee6aa85624abe3786856109453R627

I think the easiest solution is going to be just getting rid of the `not`s from that section.

Entirely my fault, sloppy merge on my part... @0rc do you have any thoughts/comments?

**Comment by dbuxton** [comment]
(Main lesson here is we need better tests ;)

**Comment by dbuxton** [comment]
I've pushed a fix in https://github.com/arachnys/cabot/commit/b468c5bf57952a3ca1459546d8d011c3ec2924f4. The root cause was a problematic PR: 41f8f4e

I have replaced `expected_num_metrics` with `allowed_num_failures` which I think is a little clearer in terms of differentiating it from `expected_num_hosts` - and now the checks work the right way around.

Let me know what you think @NickG123 and @0rc any comments welcome.
